util/store/store.go

```diff
@@ -0,0 +1,163 @@
+package store
```

Comment:
This file is an extension of `x/leverage/store.go` - I intend to remove the leverage file and switch it over to use this utility, but I'm not doing that right now so this PR stays isolated to the `x/incentive` folder.